### ISSUE TITLE
Regarding hosted content and potential preservation thereof

### ISSUE THREAD

**Issue by DramaticBox** [fastSWF/fastswf.github.io#1]
Although I understand and accept that this fastSWF is no longer supported, I also understand that quite an expanse of Flash content has been hosted on it. Thankfully, a number of efforts to archive and provide lasting access to such pieces of internet history exist, such as Flashpoint, third-party plugins, and site browsers with dedicated players. 

The question I'd like to ask is if the possibility exists to provide fastSWF's content to an 'afterlife' project such as Flashpoint, or to at least 'archive' the hosted files in some capacity, so that people may download them to play on an offline browser or provide them to an archival project on their own.

I raise this point because it just seems that, as keeper of the products of so many minds from all around the internet, there is something of a responsibility to see to the survival of their brainchildren.

**Comment by matt4530** [comment]
Hi, thanks for reaching out. We have had others reach out about preserving the content we have hosted as well, including while the site was still online, so you are not alone in this.

I'm a little hesitant in doing something about this because the users of our site have not agreed to another site hosting or archiving their content. Some users of our site, who were unable to purchase a premium plan, may have relied on the privacy of a random URL to prevent their content from being publicly available. 

So while there is good intent behind your request, I think it would be best if we relied on the content creators themselves to upload their content to an archival site so that it is preserved, if they so wish.

**Comment by DramaticBox** [comment]
That is a very good point. However, consider that a number of creators may not have access to the original files for several reasons. For example, suppose the device on which the original files were saved suffered permanent failure or its hard drive was wiped for whatever reason. Or, perhaps, they simply upgraded to a new device and forgot to carry the flash files over. Other scenarios may include corrupted files or being stored on an external media that was lost/damaged.

Whatever the case may be, there is a number of Flash files that - because their creators do not have access to them at this time - stand to cease existence entirely if not made available to at least their respective creators.

If the means exists to reach out to users, I propose two routes that might be taken: Content creators are offered to be provided with copies of their own and _only_ their own uploaded content for them to with as they please, OR users may contribute to a form expressing their consent - or lack thereof - for their work to be archived/hosted elsewhere, the data from which is used in choosing which content to provide to archivists and which content to be withheld.

tl;dr since users haven't agreed to another site hosting their content, one could either give 'em a chance to consent or help 'em get at their own content so they can upload it theirselves

**Comment by matt4530** [comment]
Yes this is a good point and we were aware that the content creators may not have had copies themselves of their content. 

For this reason, we emailed all of our registered users and all users who provided contact information when uploading content and provided a download link for all of their content. We also assisted users, via email, who had trouble retrieving their content during our site's sunsetting as well.

I don't think we plan to help or provide access to non-creators to download content that they did not create. Likewise, I don't think we plan to reach out to creators to ask if they want to archive their content somewhere. They can decide this on their own and upload their content to an archive site on their own if they so wish.
